tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/ompl/ompl_planner_configurator.h

```diff
@@ -52,20 +71,46 @@ struct OMPLPlannerConfigurator
   OMPLPlannerConfigurator(OMPLPlannerConfigurator&&) = default;
   OMPLPlannerConfigurator& operator=(OMPLPlannerConfigurator&&) = default;
 
+  /** @brief Planner type */
+  OMPLPlannerType type = OMPLPlannerType::RRTConnect;
+
   virtual ompl::base::PlannerPtr create(ompl::base::SpaceInformationPtr si) const = 0;
+
+  virtual OMPLPlannerType getType() const = 0;
+
+  virtual tinyxml2::XMLElement* toXML(tinyxml2::XMLDocument& doc) const = 0;
 };
 
 struct SBLConfigurator : public OMPLPlannerConfigurator
 {
+  SBLConfigurator() = default;
+  SBLConfigurator(const SBLConfigurator&) = default;
+  SBLConfigurator(const tinyxml2::XMLElement& xml_element);
+
+  /** @brief Planner type */
+  OMPLPlannerType type = OMPLPlannerType::SBL;
```

Comment:
I would remove this since it should never be changed and have the getType directly return the correct type in the cpp file.
```
getType() const { return OMPLPlannerType::SBL;}
```

Reply:
I would do the same for all of the others.

Reply:
That makes much more sense